### PR TITLE
Feature: Runs Filter

### PR DESCRIPTION
**Pull request by ihor-palii**
Ticket: https://app.asana.com/0/1159876841343611/1199517123604341/f

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -2198,6 +2213,35 @@ def _get_run_batches(self, flows, responded_only):
...
         if responded_only:
             runs = runs.filter(responded=True)
+
+        # filter runs by extra queries
+        extra_queries = getattr(self, "extra_queries", {})
+        if extra_queries.get("response"):
+            response_queries = extra_queries.get("response", {})
+            if response_queries.get("after"):
+                after = self.org.parse_datetime(response_queries.get("after"))
+                runs = runs.filter(created_on__gte=after)
```
Comment (teehamaral) [inline temba/flows/models.py:2223]: Per Rob recommendation, we should change from `created_on` to `modified_on` for the before/after filter

File: temba/flows/models.py [hunk 2]
```diff
@@ -2198,6 +2213,35 @@ def _get_run_batches(self, flows, responded_only):
...
+
+            if response_queries.get("before"):
+                before = self.org.parse_datetime(response_queries.get("before"))
+                runs = runs.filter(created_on__lte=before)
```
Comment (teehamaral) [inline temba/flows/models.py:2227]: Per Rob recommendation, we should change from `created_on` to `modified_on` for the before/after filter

File: temba/flows/views.py [hunk 3]
```diff
@@ -2149,11 +2166,43 @@ class RunTable(AllowOnlyActiveFlowMixin, OrgObjPermsMixin, SmartReadView):
...
+            after = self.request.GET.get("after")
+            before = self.request.GET.get("before")
+            search_query = self.request.GET.get("q")
+            contact_query = self.request.GET.get("contact_query")
+
             runs = flow.runs.all()
 
+            if after:
+                after = str_to_datetime(after, tz=flow.org.timezone, dayfirst=flow.org.date_format == "D")
+                runs = runs.filter(created_on__gte=after)
```
Comment (teehamaral) [inline temba/flows/views.py:2189]: Per Rob recommendation, we should change from `created_on` to `modified_on` for the before/after filter

File: temba/flows/views.py [hunk 4]
```diff
@@ -2149,11 +2166,43 @@ class RunTable(AllowOnlyActiveFlowMixin, OrgObjPermsMixin, SmartReadView):
...
+
+            if before:
+                before = str_to_datetime(before, tz=flow.org.timezone, dayfirst=flow.org.date_format == "D")
+                runs = runs.filter(created_on__lte=before)
```
Comment (teehamaral) [inline temba/flows/views.py:2193]: Per Rob recommendation, we should change from `created_on` to `modified_on` for the before/after filter

**Review by teehamaral — CHANGES_REQUESTED** [review]

**Comment by ihor-palii** [comment]
Updated

**Review by teehamaral — APPROVED** [review]

**Pull request merged**